content/blog/making-a-slack-bot-with-node-and-mongo/index.md

```diff
@@ -0,0 +1,623 @@
+
+![The initial screen that'll be shown when a new app is created](./initial_screen.png)
+
+This screen (and the tabs off to the side) are able to configure all of the interactions with Slack that we'll build upon with our code. We're even able to customize the look of our application in our Slack settings at the bottom of this homepage.
+
+![Towards the bottom of the intial page will show how to customize the description and such](./display_info.png)
+
+As mentioned previously, Slack provides an SDK for Node applications. [You can find the homepage for the npm package at the following URL.](https://github.com/slackapi/node-slack-sdk)
+
+In order to quickly set up the SDK, we'll create a new directory for our code to live. Once we have a clear directory, we can run:
+
+```
+npm init -y
+```
+
+To setup an initial `package.json`. Once we have a `package.json`, we can add the packages we require to use the Slack SDK:
+
+```
+npm install @slack/web-api @slack/events-api
+```
+
+After this, we'll then be able to use their example API from the README of their project as a baseline to our app:
+
+```javascript
+// index.js
+// Initialize using signing secret from environment variables
+const { createEventAdapter } = require('@slack/events-api');
+// Slack requires a secret key to run your bot code. We'll find and figure out this signing secret thing in the next steps
+const slackEvents = createEventAdapter(process.env.SLACK_SIGNING_SECRET);
+const port = process.env.PORT || 3000;
+
+// Attach listeners to events by Slack Event "type". See: https://api.slack.com/events/message.im
+slackEvents.on('message', (event) => {
+  console.log(`Received a message event: user ${event.user} in channel ${event.channel} says ${event.text}`);
+});
+
+// Handle errors (see `errorCodes` export)
+slackEvents.on('error', console.error);
+
+// Start a basic HTTP server
+slackEvents.start(port).then(() => {
+  // Listening on path '/slack/events' by default
+  console.log(`server listening on port ${port}`);
+});
+```
+
+This code is what we'll need to run a `console.log` every time a user sends a message. However, _we'll need to setup more to get this code actually running due to Slack's permissions systems_ and such. For now, we'll save this code to `index.js` in the same folder we saved our `package.json` file.
+
+Another thing that was mentioned in the code sample was the `process.env.SLACK_SIGNING_SECRET`. This is the key that Slack will use to connect your code to your workspace. We'll want to keep in mind how to store the signing secret (as the name implies, _we want to keep this key a secret as otherwise anyone can highjack your Slack app_). As the above code hints at, it's suggested to use an environmental file or configuration.
```

Comment:
```suggestion
Another thing that was mentioned in the code sample was the `process.env.SLACK_SIGNING_SECRET`. This is the key that Slack will use to connect your code to your workspace. We'll want to keep in mind how to store the signing secret (as the name implies, _we want to keep this key a secret as otherwise anyone can hijack your Slack app_). As the above code hints at, it's suggested to use an environment variable file or configuration.
```